src/pubsub/mod_pubsub_db_mnesia.erl

```diff
@@ -208,6 +210,15 @@ node_name(Nidx) ->
         _ -> <<>>
     end.
 
+-spec find_nodes_by_affiliated_user(JID :: jid:ljid()) ->
+    [{mod_pubsub:pubsubNode(), mod_pubsub:affiliation()}].
+find_nodes_by_affiliated_user(LJID) ->
```

Comment:
I actually think that such function fits better in DB layer because it should be a simple data retrieval instead the processing done in `node_flat`, which is actually a workaround for strange Mnesia data model.

Reply:
I wasn't suggesting to move it, I was just asking if there is similar functionality already there :)